### DISCUSSION TITLE
Play (possibly many) merged video and audio files via network (or use http:// in EDL)

### DISCUSSION THREAD

**Discussion by ge9** [mpv-player/mpv#15541] [Q&A]
### Expected behavior of the wanted feature

I have many pairs of audio url and video url, and want to play them as playlist.
For example, I have
https://example.com/audio?id=foo
https://example.com/video?id=foo
https://example.com/audio?id=bar
https://example.com/video?id=bar
...
And each files have title to be shown.
How can I play them as playlist?

I know there is [edl](https://github.com/mpv-player/mpv/blob/master/DOCS/edl-mpv.rst) feature, but they don't allow HTTP url. Also, there is some options like --audio-files, but, as far as I know, it can't be used for playlist like this.

**Comment by llyyr** [comment]
You can use per-file options and set the audio files via --audio-file. https://mpv.io/manual/master/#per-file-options

**Reply by ge9** [reply]
Oh, thank you. I didn't know this. But I think this is only available in comand line argument and some OSes has a limitation on command line argument length.  Also, I want to use `--prefetch-playlist` because my files are on network but its document says it may work wrongly with per-file options.

So I still want some playlist-like way to combine network audio and video.  How about creating options like `--edl-allow-http` ?

**Comment by ge9** [comment]
Now I've found a workaround to this: create a kind of "reverse-proxy" http server (e.g. `localhost:8888/audio/foo` -> `https://example.com/audio?id=foo`), mount it with rclone, and access as a local file.
EDL worked correctly then. 

However, I've noticed that the EDL file doesn't support gapless playback (usually achieved by `--prefetch-playlist`).
In my case, gapless playback is not required but desirable. 

And of course, creating such a reverse-proxy is not so easy.

So I still leave this open.
